### ISSUE TITLE
Recommend assertEqual and the other unittest assertion methods which are not deprecated in the standard library

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4989]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4989|
|Type|enhancement|
|Created|2011-03-26 19:32:50Z|
|Branch|https://github.com/twisted/twisted/tree/assert-style-recommendation-4989|

Per the mailing list discussion [TestCase methods names](http://twistedmatrix.com/pipermail/twisted-python/2011-March/023681.html), we should update our developer documentation to recommend only using `TestCase` assertion methods which the stdlib has not deprecated.

<details><summary>Searchable metadata</summary>

```
trac-id__4989 4989
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_assert_style_recommendation_4989 branches/assert-style-recommendation-4989
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1301167970000000 1301167970000000
changetime__1307109095000000 1307109095000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [31932]) Branching to 'assert-style-recommendation-4989'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

I changed the recommendation and the trial tutorial.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Yay for that, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [32023]) Merge assert-style-recommendation-4989

Author: exarkun
Reviewer: therve
Fixes: [#4989](https://github.com/twisted/twisted/issues/4989)

Change the recommendation for which assertion methods to use from the s-suffix versions
to the without-s-suffix versions (ie from `assertEquals` to `assertEqual`) to avoid a
clash with the standard library.  Also update the trial tutorial to reflect this change.
